src/hooks/useGoalActions.tsx

```diff
@@ -67,9 +74,21 @@ const useGoalActions = () => {
       }
       suggestChanges(rootGoal, goal, subGoalsHistory.length);
     } else {
-      await modifyGoal(goal.id, goal, [...ancestors, goal.id], hints);
+      const originalGoal = activeGoal;
+      if (originalGoal && hashObject(originalGoal) !== hashObject(goal)) {
```

Comment:
use activeGoal directly just add a comment that we are comparing the old and updated version.